funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -320,6 +329,22 @@ def _main_loop(self):
             num_tasks_forwarded = 0
             num_results_forwarded = 0
 
+            if self._parent_pid:
+
+                def parent_watcher(ppid: int):
+                    print(f"WATCHING PARENT: {ppid}")
+                    while ppid == os.getppid():
+                        if not self._quiesce_event.wait(timeout=1):
+                            return
+                    log.warning(f"Parent ({ppid}) has gone away.")
+                    self.stop()
+
+                threading.Thread(
+                    target=parent_watcher,
+                    args=(self._parent_pid,),  # copy; to discourage shenanigans
+                    daemon=True,
```

Comment:
Would be a good idea to name the thread.

Reply:
Added a name: `ParentPidWatcher`